connection_producer.go

```diff
@@ -115,13 +108,40 @@ func (c *redisDBConnectionProducer) Connection(ctx context.Context) (interface{}
 		return c.client, nil
 	}
 	var err error
+	var pem []byte
```

Comment:
Suggest to move this into the scope where it's used (within `if c.TLS { ... }`).